### PR TITLE
C#11 - Add test cases for Raw string literal feature

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/ControlCharacterInStringTest.cs [hunk 1]
```diff
@@ -25,19 +25,28 @@ namespace SonarAnalyzer.UnitTest.Rules
     [TestClass]
     public class ControlCharacterInStringTest
     {
+        private readonly VerifierBuilder builder = new VerifierBuilder<ControlCharacterInString>();
+
         [TestMethod]
         public void ControlCharacterInString() =>
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/ControlCharacterInStringTest.cs:31]: NITPICK: rename the test to ControlCharacterInString_CSharp8. As it is applicable only from chsarp8.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/DeadStoresTest.cs [hunk 2]
```diff
@@ -36,21 +36,36 @@ public class DeadStoresTest
 
         [TestMethod]
         public void DeadStores_SonarCfg() =>
-            sonarCfg.AddPaths("DeadStores.SonarCfg.cs").Verify();
+            sonarCfg.AddPaths("DeadStores.SonarCfg.cs")
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/DeadStoresTest.cs:39]: NITPICK:  the .Verify() invocation should go to the same line. Simple test case where it  contains only addpaths and verify we tend to put on the same line.
I am not sure it is documented anywhere (If it is not in the coding style than we should most probably document it)

Comment (csaba-sagi-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/DeadStoresTest.cs:39]: Applies for the whole PR.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/InsecureEncryptionAlgorithmTest.cs [hunk 3]
```diff
@@ -18,6 +18,8 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Security.AccessControl;
+using SonarAnalyzer.Common;
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/InsecureEncryptionAlgorithmTest.cs:22]: Unnecessary usings, please remove.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/InsecureEncryptionAlgorithmTest.cs [hunk 4]
```diff
@@ -26,27 +28,52 @@ namespace SonarAnalyzer.UnitTest.Rules
...
         [TestMethod]
         public void InsecureEncryptionAlgorithm_DoesNotRaiseIssuesForTestProject_CS() =>
-            OldVerifier.VerifyNoIssueReportedInTest(@"TestCases\InsecureEncryptionAlgorithm.cs", new CS.InsecureEncryptionAlgorithm(), GetAdditionalReferences());
+            builderCS.AddPaths("InsecureEncryptionAlgorithm.cs")
+                .AddTestReference()
+                .AddReferences(GetAdditionalReferences())
+                .VerifyNoIssueReported();
 
 #if NET
         [TestMethod]
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/InsecureEncryptionAlgorithmTest.cs:48]: NITPICK:
According to our coding style: Each compiler directive outside method body (namely #if/#endif) should be preceded and followed by an empty line.
Which means you need an empty line both there and and the endif.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/UriShouldNotBeHardcodedTest.cs [hunk 5]
```diff
@@ -33,6 +33,17 @@ public class UriShouldNotBeHardcodedTest
         public void UriShouldNotBeHardcoded_CSharp_General() =>
             builderCS.AddPaths("UriShouldNotBeHardcoded.cs").Verify();
 
+#if NET
+        [TestMethod]
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/UriShouldNotBeHardcodedTest.cs:37]: NITPICK: An empty line missing here.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/UriShouldNotBeHardcodedTest.cs [hunk 6]
```diff
@@ -33,6 +33,17 @@ public class UriShouldNotBeHardcodedTest
...
+        public void UriShouldNotBeHardcoded_CSharp11()
+        {
+            builderCS.AddPaths("UriShouldNotBeHardcoded.CSharp11.cs")
+                .WithTopLevelStatements()
+                .WithOptions(ParseOptionsHelper.FromCSharp11)
+                .Verify();
+        }
+#endif
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/UriShouldNotBeHardcodedTest.cs:45]: NITPICK: And also here.

**Comment by andrei-epure-sonarsource** [comment]
(saw this when doing backlog cleanup - please don't use acronyms like TC - we don't know what this is and the PR title will show up in the history; use words instead of acronyms)

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/SqlKeywordsDelimitedBySpaceTest.cs [hunk 7]
```diff
@@ -29,15 +29,21 @@ public class SqlKeywordsDelimitedBySpaceTest
 
         [TestMethod]
         public void SqlKeywordsDelimitedBySpace() =>
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/SqlKeywordsDelimitedBySpaceTest.cs:31]: Needs renaming to SqlKeywordsDelimitedBySpace_Csharp8

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.CSharp11.cs [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Data.Sqlite;
+
+const string part1 = """SELECT * FROM""";
+const string part2 = """ mytable WHERE mycol=""";
+const string constQuery = $"""{part1}{part2}""";
+
+void Foo(DbContext context, SqliteConnection connection, params object[] parameters)
+{
+    context.Query<User>().FromSql(constQuery, parameters); // Compliant
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/ExecutingSqlQueries.CSharp11.cs:12]: Please add a noncompliant case too, one where the strings are not constant.

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
One last round of changes.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**